src/netspresso_trainer/pipelines/base.py

```diff
@@ -262,6 +264,7 @@ def save_checkpoint(self, epoch: int):
         save_best_model = best_epoch == epoch
 
         model = self.model.module if hasattr(self.model, 'module') else self.model
+        model = copy.deepcopy(model).type(self.save_dtype)
```

Comment:
@illian01 
Do we need to copy the model if model is saved as fp32?
This may waste memory if the models are with fp32, which are most cases.
Our models are mostly small, so it does not problem right now, but when the model comes to be large, we might need to think about the memory.

Reply:
@deepkyu 
As you mentioned, I thought it's okay because our models are mostly small.
I agree about that we have to consider the memory for the future.

I'll change it to do different action according to `self.save_dtype`